docs(devtools): write feature descriptions for new DI debugging features in Angular Devtools

AleksanderBodurri opened the pull request:
This commit adds sections to the devtools aio page for the new DI debugging features in Angular DevTools:

1. Directive dependency inspection
2. Injector hierarchy visualization

dgp1130 commented on aio/content/guide/devtools.md:
```diff
@@ -105,6 +105,17 @@ After you select a particular component, click the icon at the top-right of the
 </div>
 
+### View injected service dependencies of components
+Starting in Angular 17, component dependencies are viewable in the property viewer. After you select a particular component, if that component has dependencies, you'll be able to see them listed under the *"Injected Services"* bar.
```
Nit: "Component dependencies" is kind of vague and makes me think of other components which get imported. I'd maybe go with "Injected Services" like it's actually labeled.

Question: They don't need to be "services" right? It could be any injected value? Maybe that's too precise to worry about here, but maybe "Injected tokens" might be clearer?

dgp1130 commented on aio/content/guide/devtools.md:
```diff
@@ -105,6 +105,17 @@ After you select a particular component, click the icon at the top-right of the
+
+<div class="lightbox">
+
+<img alt="An image showing how injected services are visible when a component is selected in the components tab." src="generated/images/guide/devtools/di-component-deps.png">
```
Nit: IIUC screen readers will state that you're looking at an image before reading the `alt`, so starting with "An image" is redundant. Maybe something like "A screenshot of Angular DevTools components tab showing injected services for a selected component." would provide more meaningful information?

Same suggestion for the below images.

dgp1130 commented on aio/content/guide/devtools.md:
```diff
@@ -242,6 +253,43 @@ Then, import the file in the initial view of the profiler by clicking the **Choo
+
+
+
+
```
Nit: Can we trim out some of the unnecessary whitespace here? I'm kinda surprised the formatter didn't clean it up automatically.

dgp1130 commented on aio/content/guide/devtools.md:
```diff
@@ -242,6 +253,43 @@ Then, import the file in the initial view of the profiler by clicking the **Choo
+
+### View the injector hierarchy of your application
+
+The **Injector Tree** tab lets you explore the structure of the Injectors configured in your application. Here you will see two trees representing the [injector hiearchy](guide/hierarchical-dependency-injection) of your application. One tree is your environment hierarchy, the other is your element hierachy.
```
Suggestion: We should give some more context about the two hierarchies and what they represent. If there's any existing docs covering this in dependency injection, we should definitely link to those.

AleksanderBodurri committed:
docs(devtools): write feature descriptions for new DI debugging features in Angular Devtools

This commit adds new sections to the devtools aio page for the new DI debugging features in Angular DevTools:

1. Directive dependency inspection
2. Injector hierarchy visualization

dgp1130 commented on aio/content/guide/devtools.md:
```diff
@@ -105,6 +105,17 @@ After you select a particular component, click the icon at the top-right of the
 </div>
 
+### View injected services of components
+Starting in Angular 17, services that are injected in a component or directive context are viewable in the property viewer. After you select a particular component, if that component has dependencies, you'll be able to see them listed under the *"Injected Services"* bar.
```
Voice: Passive voice, contraction.

> After you select a particular component, the *"Injected Services"* bar will list them.

dgp1130 commented on aio/content/guide/devtools.md:
```diff
@@ -105,6 +105,17 @@ After you select a particular component, click the icon at the top-right of the
+### View injected services of components
+Starting in Angular 17, services that are injected in a component or directive context are viewable in the property viewer. After you select a particular component, if that component has dependencies, you'll be able to see them listed under the *"Injected Services"* bar.
+
+By clicking on a service, an expansional panel will appear that visualizes the resolution path that Angular used to resolve that service.
```
Typo / passive voice:

> By clicking on a service, an ~expansional~ **expansion** panel will appear that visualizes the resolution path ~that~ Angular used to resolve that service.

dgp1130 commented on aio/content/guide/devtools.md:
```diff
@@ -242,10 +253,43 @@ Then, import the file in the initial view of the profiler by clicking the **Choo
+
+### Visualize resolution paths
+
+When a specific injector is selected, the path that Angular's depenedency injection algorithm traverses from that injector to the root is highlighted. For element injectors, this includes highlighting the environment injectors that the dependency injection algorithm jumps to when a dependency cannot be resolved in the element hierarchy.
```
I think [resolution rules](https://angular.io/guide/hierarchical-dependency-injection#resolution-rules) has some really good content about how to think about these trees and the resolution path. Can we fit in a link to that somewhere in here?

dgp1130 commented on aio/content/images/guide/devtools/di-component-deps.png:
Suggestion: The text on these images is pretty small and can be bad for accessibility. AIO in particular doesn't do a great job of making the images zoomable. Even I'm having trouble reading anything in the last screenshot.

I suggest zooming DevTools itself (I think Ctrl+Scroll will do this) before taking the screenshot so the text is more legible in these screenshots.

dgp1130 reviewed (commented):
Ah didn't realize we didn't merge this prior to v17 release. Mostly some minor wording nitpicks. Also I didn't realize the images were a bit difficult to read, so anything we can do to make them more legible would be really helpful here.

dylhunn commented:
This PR was merged into the repository by commit 7252ee41621c2149fd284a683b94dc8f1beb6b19.
